### ISSUE TITLE
Sending email fails if `firstname` or `lastname` fields are disabled

### ISSUE THREAD

**Issue by bradjones1** [mautic/mautic#11807]
### Mautic Version

4.4.x series

### PHP version

8.0.26

### What browsers are you seeing the problem on?

Not relevant

### What happened?

I do not collect/have first or last names for contacts, nor most of the other default fields in Mautic, so I disabled them. Turns out that depending on your configuration, the missing array keys at https://github.com/mautic/mautic/blob/757d16ed7dfed55e5d6e0a238063eb37af5fa253/app/bundles/EmailBundle/Model/SendEmailToContact.php#L198-L201 will throw an exception, and the email will not send, nor will the `::failContact()` method run.

### How can we reproduce this issue?

1. Disable `firstname` and/or `lastname` fields.
2. Try to send the contact an email, e.g. during a campaign.

### Relevant log output

```shell
cron_1      | {"message":"Error thrown while running command \"{command}\". Message: \"{message}\"","context":{"exception":{"class":"ErrorException","message":"Warning: Undefined array key \"firstname\"","code":0,"file":"/var/www/html/docroot/app/bundles/EmailBundle/Model/SendEmailToContact.php:199"},"command":"mautic:campaigns:trigger","message":"Warning: Undefined array key \"firstname\""},"level":500,"level_name":"CRITICAL","channel":"console","datetime":{"date":"2022-12-12 20:45:06.251972","timezone_type":3,"timezone":"America/Denver"},"extra":{"hostname":"f76c8a7ef301","pid":2476}}
cron_1      | {"message":"Command `mautic:campaigns:trigger` exited with status code 1","context":[],"level":300,"level_name":"WARNING","channel":"mautic","datetime":{"date":"2022-12-12 20:45:06.254727","timezone_type":3,"timezone":"America/Denver"},"extra":{"hostname":"f76c8a7ef301","pid":2476}}
```
```


### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by bradjones1** [comment]
And when the environment is `prod`, a bit different:

```
{"message":"CAMPAIGN: Failed to start the session because headers have already been sent by \"/var/www/html/docroot/app/bundles/EmailBundle/Model/SendEmailToContact.php\" at line 199 in . (which is loaded in resource \".\"). Make sure there is a loader supporting the \"mautic\" type.","context":[],"level":400,"level_name":"ERROR","channel":"mautic","datetime":{"date":"2022-12-12 21:06:09.103940","timezone_type":3,"timezone":"America/Denver"},"extra":{"hostname":"d0dd8b8d8072","pid":527}}
```

**Comment by RCheesley** [comment]
Thanks for the report @bradjones1 - as with the other ones, if you're open to making a PR here's the docs on how to get started: https://contribute.mautic.org/contributing-to-mautic/developer

**Comment by bradjones1** [comment]
I definitely intend to loop around on some of these issues with PRs. I've been busy implementing Mautic for my startup and trying to document the pain points and "WTF?" moments as I go.

**Comment by mabumusa1** [comment]
This issue is solved with #11613

**Comment by bradjones1** [comment]
Not stale.
